### PR TITLE
Stats: Updating UTM module with empty states

### PR DESCRIPTION
**Pull request by grzegorz-cp**
Related to https://github.com/Automattic/red-team/issues/72

## Proposed Changes

* adding new empty state for UTM
* extending UTM Builder to accept a custom trigger
* unity TS types for the new modules

SCR-20240702-mpbt

## Why are these changes being made?

* a part of the empty states project

## Testing Instructions

* go to the live branch
* apply `stats/empty-module-traffic` feature flag
* verify that a page with an empty UTM component works with and without the feature flag
* repeat for blogs with data
* verify that the advanced feature overlay works properly
* verify that the new empty state only shows up after applying the feature flag

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/stats/stats-module-utm-builder/stats-module-utm-builder.tsx [hunk 1]
```diff
@@ -26,14 +27,22 @@ const UTMBuilder: React.FC< Props > = ( { modalClassName } ) => {
...
-				className="stats-utm-builder__trigger"
-				onClick={ handleClick }
-				variant="secondary"
-			>
-				{ translate( 'URL Builder' ) }
-			</Button>
+			{ /* div wrapper is needed to support custom trigger elements with an interal click event */ }
+			{ trigger ? (
+				// eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/no-noninteractive-tabindex
+				<div onClick={ handleClick } onKeyDown={ handleClick } tabIndex={ 0 }>
```
Comment (grzegorz-cp) [inline client/my-sites/stats/stats-module-utm-builder/stats-module-utm-builder.tsx]: I'm on the fence if this is a good way around it or cloning an element and passing an internal onClick would be better. Now I'm leaning toward the other way.

**Comment by grzegorz-cp** [comment]
I need to include a feature flag so it's back to a draft.

File: client/my-sites/stats/stats-module-utm-builder/stats-module-utm-builder.tsx (hunk 1 above)
Comment (grzegorz-cp) [reply client/my-sites/stats/stats-module-utm-builder/stats-module-utm-builder.tsx]: I replaced this.

**Review by annacmc — COMMENTED** [review]
This looks great on my site with UTM builder active:

image


But I'm getting some weird layout stuff going on in this branch without the feature flag on the same site, where the UTM builder is being hidden and the layouts are not adjusting well around it:

image

**Comment by grzegorz-cp** [comment]
Thanks for catching this @annacmc. I've just pushed a change to show the module when the feature flag is not enabled.

**Comment by grzegorz-cp** [comment]
It looks good to me now. I tested with and without feature flag:
   - main page
   - summary page
   - post details page
   - different environments

**Review by annacmc — APPROVED** [review]
LGTM 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16022038

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @grzegorz-cp for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16022038).

**Pull request merged**